sklearn/ensemble/tests/test_forest.py

```diff
@@ -827,6 +832,23 @@ def test_sparse_input():
         yield check_sparse_input, name, X, sparse_matrix(X), y
 
 
+def check_no_sparse_y_support_classifier(name):
+    X, y = datasets.make_multilabel_classification(random_state=0,
+                                                   n_samples=50)
+    y_sparse = csr_matrix(y)
+    ForestClassifier = FOREST_CLASSIFIERS[name]
+    assert_raise_message(TypeError, 'A sparse matrix was passed, but '
+                         'dense data is required. Use X.toarray() to '
```

Comment:
Yes, I agree that is an issue. But is there any other alternative to this because at the end we are relying on `check_array` to raise the error message?